internal/vm/ioproxy.go

```diff
@@ -129,7 +130,11 @@ func (connectorPair *IOConnectorPair) proxy(
 
 		_, err := io.CopyBuffer(writer, reader, make([]byte, internal.DefaultBufferSize))
 		if err != nil {
-			logger.WithError(err).Error("error copying io")
+			if strings.Contains(err.Error(), "use of closed network connection") {
```

Comment:
While Go would [keep this string as a public interface](https://github.com/golang/go/blob/bdea352b96c6934ee49920b18c3b8aedffd27d08/src/internal/poll/fd.go#L20), checking the error by substring match doesn't seem right.

Can't we let `ioproxy` know that the status of the connection, rather than ignoring the error?

Reply:
@sipsma What do you think?

Reply:
We are just operating on generic `io.Reader`s and `io.Writer`s, which is what allows this code to work equivalently across permutations of vsocks, unix sockets and fifos, so I'm not sure there's a straightforward way to just check the status of the connection as the "connection" could be any number of things.

I really don't like the whole method of doing error checking by string matching, but it's basically accepted as the way you have to do it in go at the moment (if I'm wrong or missing something please let me know). There's some hope with the recent error improvements in go 1.13, but to my knowledge that hasn't had enough time to soak into the various interfaces we use here, so for now string matching is okay with me.